$OpenBSD: patch-plasma_generic_dataengines_geolocation_location_gps_cpp,v 1.2 2020/05/24 06:06:06 kirby Exp $

fix build with newer gpsd API
upstream commit 81a0fe6db9ac9044c2ec8a246878ec65651cb6dd
and 0c4974d68804cdaff2efb6317f7853a89d3a3d2b

Index: plasma/generic/dataengines/geolocation/location_gps.cpp
--- plasma/generic/dataengines/geolocation/location_gps.cpp.orig
+++ plasma/generic/dataengines/geolocation/location_gps.cpp
@@ -51,13 +51,19 @@ void Gpsd::run()
     while (!m_abort) {
         Plasma::DataEngine::Data d;
 
-#if GPSD_API_MAJOR_VERSION >= 5
+#if GPSD_API_MAJOR_VERSION >= 7
+	if (gps_read(m_gpsdata, NULL, 0) != -1) {
+#elif GPSD_API_MAJOR_VERSION >= 5
 	if (gps_read(m_gpsdata) != -1) {
 #else
         if (gps_poll(m_gpsdata) != -1) {
 #endif
             //kDebug() << "poll ok";
+#if GPSD_API_MAJOR_VERSION >= 9
+            if (m_gpsdata->online.tv_sec || m_gpsdata->online.tv_nsec) {
+#else
             if (m_gpsdata->online) {
+#endif
                 //kDebug() << "online";
                 if (m_gpsdata->status != STATUS_NO_FIX) {
                     //kDebug() << "fix";
